src/django/pfb_analysis/models.py

```diff
@@ -93,10 +95,43 @@ class Status(object):
     neighborhood = models.ForeignKey(Neighborhood,
                                      related_name='analysis_jobs',
                                      on_delete=models.CASCADE)
-    status = models.CharField(choices=Status.CHOICES,
-                              default=Status.CREATED,
-                              max_length=12,
-                              help_text='The current status of the AnalysisJob')
+
+    @property
+    def status(self):
+        """ Return current status for this job
+
+        Combines the sources of status information (updates, batch job status, inference) to
+        return the best answer.
+        """
+        # If there's no job ID that means it's brand new
+        if self.batch_job_id is None:
+            return self.Status.CREATED
+
+        if self.batch_job_status == 'FAILED':
+            return self.Status.ERROR
+        try:
+            # If the job hasn't failed and has sent any status updates, use the latest
+            return self.status_updates.last().status
+        except AttributeError:
+            # Not failed but no status updates means it must be in the queue
+            return self.batch_job_status
```

Comment:
Yeah, I think you're right that using the batch job status as the status isn't great.  It would make `AnalysisJob.Status` an incomplete list, with the extra members being outside the app and beyond our control.  I'll add `QUEUED` and use it for anything that's been sent (i.e. has a batch ID) but hasn't provided updates yet.  I think it still makes sense to check `FAILED`, though, since a job that crashes won't send a status update to tell us so, so it would look like it was stuck on the last milestone it successfully passed.